src/lib/builders/table-of-contents/create.ts

```diff
@@ -0,0 +1,259 @@
+import { addEventListener, builder, createElHelpers, executeCallbacks, generateId, isBrowser } from "$lib/internal/helpers";
+import type { Defaults } from "$lib/internal/types";
+
+import { onMount } from "svelte";
+import { get, writable, type Writable } from "svelte/store";
+
+import type { Heading, CreateTableOfContentsArgs, ElementHeadingLU, HeadingParentsLU, TableOfContentsItem, NavItemProps } from "./types";
+
+const defaults = {
+    exclude: ['h1'],
+    scrollOffset: 0,
+    scrollBehaviour: 'smooth',
+    tocType: 'lowest',
+} satisfies Defaults<CreateTableOfContentsArgs>;
+
+// Do we need this?
+// const { name } = createElHelpers('table-of-contents');
```

Comment:
Always, if we're exposing stores that contain element props. In this case you'd use it in item

Reply:
Great! I'm still getting familiar with how the builders API works 😄 I'll add it and will let you know if I have questions about it.